devbot/tools/delay.py

```diff
@@ -0,0 +1,49 @@
+""" A helper tool to delay messages or schedule messages """
+
+import asyncio
+import datetime
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def delay_message(delay, message):
+    """ Returns the message after delay or, if delay is of type datetime/time at that time. """
+    # If delay is a datetime return at given time
+    if isinstance(delay, datetime.datetime):
+        if delay < datetime.datetime.now():
+            return "Time is in the past dummy."
+        delta = delay - datetime.datetime.now()
+        seconds = delta.total_seconds()
+        LOGGER.debug(f"Scheduled to return {message} at {delay}")
+        await asyncio.sleep(seconds)
+        LOGGER.debug(f"Returning {message} after {delay}")
+        return message
+    # If delay is time return at time on same date
+    elif isinstance(delay, datetime.time):
+        if delay < datetime.datetime.now().time():
+            return "Time is in the past dummy."
+        delta = delay - datetime.datetime.now().time()
+        seconds = delta.total_seconds()
+        LOGGER.debug(f"Scheduled to return {message} at {delay}")
+        await asyncio.sleep(seconds)
+        LOGGER.debug(f"Returning {message} after {delay}")
+        return message
+    # If delay is timedelta
+    elif isinstance(delay, datetime.timedelta):
+        seconds = delay.total_seconds()
+        LOGGER.debug(f"Scheduled to return {message} after {seconds} seconds")
+        await asyncio.sleep(seconds)
+        LOGGER.debug(f"Returning {message} after {seconds} seconds")
```

Comment:
"Returning {message} after {seconds} seconds" is an ambiguous log entry. Changing "Returning" to "Returned" might be better, but then you have the problem that you log the return before it has actually happened. Hmmmmmmm.